lib/Chat/ReactionManager.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2021 Vitor Mattos <vitor@php.rio>
+ *
+ * @author Vitor Mattos <vitor@php.rio>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Chat;
+
+use OCA\Talk\Participant;
+use OCA\Talk\Room;
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\Comments\IComment;
+use OCP\Comments\ICommentsManager;
+use OCP\Comments\NotFoundException;
+use OCP\IL10N;
+
+class ReactionManager {
+	/** @var ICommentsManager|CommentsManager */
+	private $commentsManager;
+	/** @var IL10N */
+	private $l;
+	/** @var MessageParser */
+	private $messageParser;
+	/** @var ITimeFactory */
+	protected $timeFactory;
+
+	public function __construct(CommentsManager $commentsManager,
+								IL10N $l,
+								MessageParser $messageParser,
+								ITimeFactory $timeFactory) {
+		$this->commentsManager = $commentsManager;
+		$this->l = $l;
+		$this->messageParser = $messageParser;
+		$this->timeFactory = $timeFactory;
+	}
+
+	public function addReactionMessage(Room $chat, Participant $participant, int $messageId, string $reaction): IComment {
+		$comment = $this->commentsManager->create(
+			$participant->getAttendee()->getActorType(),
+			$participant->getAttendee()->getActorId(),
+			'chat',
+			(string) $chat->getId()
+		);
+		$comment->setParentId((string) $messageId);
+		$comment->setMessage($reaction);
+		$comment->setVerb('reaction');
+		$this->commentsManager->save($comment);
```

Comment:
Since this is all bypassing all the chat manager code, there are no notifications, is that intended?
I think it's kind of important (at least in one-to-one conversations) to be notified so you know when people voted.

Reply:
As per call let's notify users based on the "notify for every chat message" setting